pkg/apis/azure/validation/infrastructure.go

```diff
@@ -48,13 +48,10 @@ func ValidateInfrastructureConfig(infra *apisazure.InfrastructureConfig, resourc
 	// validation here will fail for those cases.
 	// TODO: remove the following block and uncomment below blocks once deployment into existing resource groups works properly.
 	if infra.ResourceGroup != nil {
-		allErrs = append(allErrs, field.Invalid(field.NewPath("resourceGroup"), infra.ResourceGroup, "specifying an existing resource group is not supported yet"))
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("resourceGroup"), infra.ResourceGroup, "specifying an existing resource group is not supported yet"))
 	}
 
-	networksPath := field.NewPath("networks")
-	if len(infra.Networks.Workers) == 0 {
```

Comment:
Why was this removed?

Reply:
`cidrvalidation.NewCIDR` in line 56 already validates the correct CIDR range and an empty string is also invalid. We can also do it explicitly if you if you prefer the way it was done before?

Reply:
The point is that before the user got two error messages:

```
              {
                  Type: "FieldValueRequired",
                  Field: "networks.workers",
                  BadValue: "",
                  Detail: "must specify the network range for the worker network",
              },
              {
                  Type: "FieldValueInvalid",
                  Field: "networks.workers",
                  BadValue: "",
                  Detail: "invalid CIDR address: ",
              },
```

So I thought it would be sufficient to only return `invalid CIDR address`.